uniffi/src/templates/TopLevelFunctionTemplate.rs

```diff
@@ -4,23 +4,13 @@
 // send data across the FFI, which will fail to compile if the provided function does not match what's
 // specified in the IDL.    
 #}
-{%- match func.ffi_func().return_type() -%}
-{%- when Some with (return_type) %}
 
 #[no_mangle]
 pub extern "C" fn {{ func.ffi_func().name() }}(
-    {%- call rs::arg_list_rs_decl(func.ffi_func().arguments()) %}) -> {{ return_type|ret_type_c }} {
+    {% call rs::arg_list_rs_decl(func.ffi_func()) %}
+) -> {% call rs::return_type_func(func) %} {
+    // If the provided function does not match the signature specified in the IDL
+    // then this attempt to cal it will not compile, and will give guideance as to why.
```

Comment:
```suggestion
    // then this attempt to call it will not compile, and will give guidance as to why.
```

Side note, I really like how the `to_rs_function_call` macro spells out `<T as uniffi::support::ViaFfi>::try_from_ffi`, so if you're trying to send an unsupported type or pass it at the wrong position, it'll give you a useful error! 👍